docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+    import "https://example.com/token.sol"; // source unit name: https://example.com/token.sol
+
+The import path translates directly to a source unit name without normalization of any kind:
+
+::
+
+    import "/project/lib/../lib///math.sol"; // source unit name: /project/lib/../lib///math.sol
+    import "lib/../lib///math.sol";          // source unit name: lib/../lib///math.sol
+
+In the above you might expect the source unit names to be ``/project/lib/math.sol`` and
+``lib/math.sol`` respectively but this is not the case.
+For direct imports the source unit name is exactly what is stated in the import.
+When the source is provided via Standard JSON interface each of these names can actually be
+associated with different content.
+
+When the source is not available in the virtual filesystem, the compiler passes the source unit name
+to the file loader.
+The default loader will attempt to use it as a path and look up the file on disk.
+At this point the platform-specific normalization rules kick in and ``/project/lib/math.sol`` and
+``/project/lib/../lib///math.sol`` may actually result in the same file being loaded.
+Note, however, that the compiler will still see them as separate source units that just happen to
+have identical content.
+
+.. note::
+
+    While the rules for translating import paths into source unit names are the same on every
+    platform, the default file loader uses platform-specific rules to locate files on disk.
+    This means that for example this import might result in the file being successfully loaded from
+    disk when compiling on Windows but not on other platforms:
+
+    .. code-block:: solidity
+
+        import "C:\\project\\lib\\token.sol"; // source unit name: C:\project\lib\token.sol
+
+    To compile such a project on a different platform you would have to use the Standard JSON
+    interface and provide the source directly under the right source unit name.
+    For this reason relying on platform-specific behaviour of the file loader is highly discouraged.
+
+.. index:: ! relative import, ! import; relative
+.. _relative-imports:
+
+Relative Imports
+================
+
+An import starting with ``./`` or ``../`` is a *relative import*.
+Such imports specify the path relative to the source unit name of the importing source unit:
+
+.. code-block:: solidity
+    :caption: /project/lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: /project/lib/util.sol
+    import "../token.sol" as token; // source unit name: /project/token.sol
+
+.. code-block:: solidity
+    :caption: lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: lib/util.sol
+    import "../token.sol" as token; // source unit name: token.sol
+
+.. note::
+
+    Do not confuse relative imports with relative paths.
+    Both ``util.sol`` and ``./util.sol`` specify relative paths on disk but these paths are treated
```

Comment:
But this depends on the compiler binary. This section is rather confusing - maybe  we should clarify it as early as possible and talk about "filesystem paths relative to the current working directory"

Reply:
I'm leaving it here for now but I'll probably rewrite it for the blog post anyway. In #11442 I removed it almost completely and just added a short note.